include/manifold/linalg.h

```diff
@@ -828,10 +841,10 @@ struct vec<T, 4> {
       : vec(xyz.x, xyz.y, xyz.z, w_) {}
   constexpr explicit vec(const T &s) : vec(s, s, s, s) {}
   constexpr explicit vec(const T *p) : vec(p[0], p[1], p[2], p[3]) {}
-  template <class U>
-  constexpr explicit vec(const vec<U, 4> &v)
-      : vec(static_cast<T>(v.x), static_cast<T>(v.y), static_cast<T>(v.z),
-            static_cast<T>(v.w)) {}
+  template <class U, int N>
+  constexpr explicit vec(const vec<U, N> &v)
+      : vec(static_cast<T>(v.x), N < 2 ? 0 : static_cast<T>(v.y),
+            N < 3 ? 0 : static_cast<T>(v.z), N < 4 ? 0 : static_cast<T>(v.w)) {}
```

Comment:
I changed their constructor to allow it to take vectors of different lengths like GLM does.

Reply:
Does anyone know a nice templatey way to allow these to take a wider variety of inputs a la GLM, e.g. `vec3(double, vec2)` or `vec4(vec3, double)`?

Reply:
Actually, linalg already has the second type - that's probably good enough. What I really need is `mat4(mat3)` and `mat3(mat4x3)` etc.

Reply:
Okay, I think we're good here now too.